platform/security/certificate/security-crl-generator/src/main/java/org/codice/ddf/security/crl/generator/CrlGenerator.java

```diff
@@ -133,36 +136,34 @@ public synchronized void run() {
       return;
     }
 
-    FileBackedOutputStream fileBackedOutputStream = null;
-    try {
-      // Read the response content and get the byte source
-      ByteSource byteSource;
-      try (InputStream entityStream = (InputStream) entity) {
-        fileBackedOutputStream = new FileBackedOutputStream(FILE_BACKED_STREAM_THRESHOLD);
+    try (FileBackedOutputStream fileBackedOutputStream =
+            new FileBackedOutputStream(FILE_BACKED_STREAM_THRESHOLD);
+        InputStream entityStream = (InputStream) entity) {
+      try {
+        // Read the response content and get the byte source
         IOUtils.copy(entityStream, fileBackedOutputStream);
-        fileBackedOutputStream.close();
-        byteSource = fileBackedOutputStream.asByteSource();
-      }
+        ByteSource byteSource = fileBackedOutputStream.asByteSource();
 
-      File crlFile = getCrlFile(byteSource);
-      // Verify that the CRL is valid
-      if (!crlIsValid(byteSource)) {
-        postErrorEvent("An error occurred while validating the CRL.");
-        return;
-      }
-      writeCrlToFile(byteSource, crlFile);
-    } catch (IOException e) {
-      LOGGER.warn("Unable to copy CRL to local CRL. {}", e.getMessage());
-      postErrorEvent("An error occurred while downloading the CRL.");
-    } finally {
-      // Cleanup temp file
-      if (fileBackedOutputStream != null) {
+        File crlFile = getCrlFile(byteSource);
+        // Verify that the CRL is valid
+        if (!crlIsValid(byteSource)) {
+          postErrorEvent("An error occurred while validating the CRL.");
+          return;
+        }
+        writeCrlToFile(byteSource, crlFile);
+      } catch (Exception e) {
+        LOGGER.warn("Unable to copy CRL to local CRL. {}", e.getMessage());
+        postErrorEvent("An error occurred while downloading the CRL.");
+      } finally {
+        // Cleanup temp file
         try {
           fileBackedOutputStream.reset();
         } catch (IOException e) {
           LOGGER.warn("Error occurred while deleting the temporary file. {}", e.getMessage());
```

Comment:
✏️ You can leave it if you want the extra specificity in the logging, but it might be a bit cleaner to just do `fileBackedOutputStream.reset()` in here and let the exception be caught by L165 (`catch (Exception e)`)

Reply:
I think it is better to keep it for specificity as a download error message would be useful for quickly diagnosing download issues that are likely to be more common than other types.